### ISSUE TITLE
ERROR - garbageHandler._collectionCallback - Found at least 101 unreachable objects in run - after closing outlook 365 and moving to the system tray

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#11766]
### Steps to reproduce:
This issue does not reproduce very reliably. It occured when I closed MS outlook 365 and moved to the system tray to check the time.

### Actual behavior:
Following warnings and error is occuring:

```
IO - inputCore.InputManager.executeGesture (22:50:35.163) - winInputHook (13488):
Input: kb(laptop):upArrow
IO - inputCore.logTimeSinceInput (22:50:35.219) - MainThread (14020):
0.056 sec since input
IO - speech.speak (22:50:35.219) - MainThread (14020):
Speaking ['Systemuhr, 22:50, 15.10.2020', 'Schalter']
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.734) - MainThread (14020):
Garbage collector has found one or more unreachable objects. See further warnings for specific objects.
Stack trace:
  File "nvda.pyw", line 236, in <module>
  File "core.pyc", line 553, in main
  File "wx\core.pyc", line 2134, in MainLoop
  File "gui\__init__.pyc", line 1062, in Notify
  File "core.pyc", line 522, in run
  File "IAccessibleHandler\__init__.pyc", line 1094, in pumpAll
  File "IAccessibleHandler\__init__.pyc", line 934, in processShowWinEvent
  File "IAccessibleHandler\__init__.pyc", line 636, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible\__init__.pyc", line 48, in getNVDAObjectFromEvent
  File "NVDAObjects\__init__.pyc", line 116, in __call__
  File "abc.pyc", line 143, in __subclasscheck__
  File "abc.pyc", line 143, in __subclasscheck__
  File "abc.pyc", line 143, in __subclasscheck__
  [Previous line repeated 1 more time]
  File "garbageHandler.pyc", line 23, in __del__
  File "garbageHandler.pyc", line 64, in notifyObjectDeletion
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.734) - MainThread (14020):
Deleting unreachable object <'outlook' (appName 'outlook', process ID 11372) at address 86abf10>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.734) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x05E25AB0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.735) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x086AB8D0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.735) - MainThread (14020):
Deleting unreachable object <NVDAObjects.window.Desktop object at 0x03861790>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.735) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6019fa8 at c526260>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.735) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x601a160 at c526a30>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.735) - MainThread (14020):
Deleting unreachable object <appModules.outlook.OutlookWordDocument object at 0x09BB0A70>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.735) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeComp) ptr=0x61247a4 at c526da0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x0C500690>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.winword.ProtectedDocumentPane object at 0x0C7FCD70>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x0C7FCD10>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x086ABB90>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x09BB0870>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x09BB0750>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x09BB08D0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x09BB09D0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.736) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x09BB07B0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.737) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x601a108 at c526800>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.737) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x09BB04B0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.737) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6019be0 at c526a80>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.737) - MainThread (14020):
Deleting unreachable object <POINTER(IDispatch) ptr=0x612424c at c5269e0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.737) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeInfo) ptr=0x612475c at c5267b0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x601a210 at c526e90>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6019a28 at c5264e0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6019a80 at c526cb0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x601a478 at c526760>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x601a2c0 at c526850>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6019ad8 at c5268a0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6019c90 at c526b70>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.738) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6019b30 at c5268f0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6124324 at c526e40>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x09B68AD0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x09B9EDF0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <appModules.outlook.MailViewerTreeInterceptor object at 0x09B9EDD0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeComp) ptr=0x6124c24 at c526d50>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeComp) ptr=0x6125014 at c526940>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x601a000 at c526710>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x601a268 at c526df0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.739) - MainThread (14020):
Deleting unreachable object <POINTER(IDispatch) ptr=0x6124834 at c526300>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.740) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeInfo) ptr=0x6124b4c at c5265d0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.740) - MainThread (14020):
Deleting unreachable object <POINTER(IDispatch) ptr=0x6124cfc at c526f30>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.740) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeInfo) ptr=0x6124e1c at c526d00>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.740) - MainThread (14020):
Deleting unreachable object <'appModuleHandler' (appName 'chrome', process ID 12744) at address c7f0c10>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.740) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.chromium.ChromeVBuf object at 0x09BCDB10>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.chromium.Document object at 0x0C7F0990>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible2) ptr=0x612664c at 9bbe5d0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleAction) ptr=0x61267fc at 9bbe3f0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleText) ptr=0x6126b14 at 9bbee40>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x049A3150>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x61257ac at 9bbe670>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x086AB030>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible2) ptr=0x6126064 at 9bbe530>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleText) ptr=0x61269f4 at 9bbe3a0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.741) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeComp) ptr=0x61c48b4 at 9bbea80>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.742) - MainThread (14020):
Deleting unreachable object <POINTER(IDispatch) ptr=0x61c462c at 9bbe800>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.742) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeInfo) ptr=0x61c4824 at 9bbe760>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.742) - MainThread (14020):
Deleting unreachable object <appModules.outlook.MailViewerTextInfoForTreeInterceptor object at 0x09BB0A10>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.742) - MainThread (14020):
Deleting unreachable object <NVDAObjects.window.winword.WordDocumentTextInfo object at 0x09BE89B0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.742) - MainThread (14020):
Deleting unreachable object <POINTER(IDispatch) ptr=0x148b0c4 at 9bbe490>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.743) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeInfo) ptr=0x61250ec at 9bbe6c0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.743) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeComp) ptr=0x61249e4 at 9bbe7b0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.743) - MainThread (14020):
Deleting unreachable object <POINTER(IDispatch) ptr=0x148b154 at 9bbe8f0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.743) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeInfo) ptr=0x61250ec at 9be5940>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.743) - MainThread (14020):
Deleting unreachable object <POINTER(ITypeComp) ptr=0x61249e4 at 9be5e40>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.743) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x09B9EFB0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.743) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible2) ptr=0x6126064 at 9bbe710>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleText) ptr=0x61269f4 at c51c8f0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x0421EA30>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6126bec at 9bbe620>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <NVDAObjects.window.Desktop object at 0x0C7F0B30>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible2) ptr=0x6126064 at 9be5f80>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x09BE8110>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleText) ptr=0x61269f4 at 9be5440>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.744) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6127da4 at 9bbe990>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x09BE8D90>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <NVDAObjects.window.Desktop object at 0x049A3130>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible2) ptr=0x6126064 at 9bbec10>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x0C7F0B90>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleText) ptr=0x61269f4 at 9be5580>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible2) ptr=0x612664c at 9be5350>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.chromium.Document object at 0x0C5000D0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleAction) ptr=0x61267fc at 9be50d0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.745) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleText) ptr=0x6126b14 at 9be5cb0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.746) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x61257ac at 9be5bc0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.746) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x09C07710>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.746) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible2) ptr=0x6126064 at 9be5170>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.746) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x09BE8ED0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.746) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessibleText) ptr=0x61269f4 at 9be58f0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6159be0 at c526b20>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x086BBDB0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <POINTER(IUIAutomationElement) ptr=0x6127bf0 at 9bbe0d0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <appModules.outlook.UIAGridRow object at 0x09B9ED50>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x61fd550 at 9bbea30>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.WindowRoot object at 0x09BB03F0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <NVDAObjects.window.Desktop object at 0x079BFFD0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x61fcf78 at 9bbe850>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x076EFC70>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.747) - MainThread (14020):
Deleting unreachable object <POINTER(IUIAutomationElement) ptr=0x6125688 at 797cd50>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.748) - MainThread (14020):
Deleting unreachable object <NVDAObjects.UIA.UIA object at 0x09BE2490>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.748) - MainThread (14020):
Deleting unreachable object <POINTER(IAccessible) ptr=0x6159c38 at 9bbeda0>
WARNING - garbageHandler.notifyObjectDeletion (22:50:35.748) - MainThread (14020):
Deleting unreachable object <NVDAObjects.IAccessible.IAccessible object at 0x09C077B0>
ERROR - garbageHandler._collectionCallback (22:50:35.748) - MainThread (14020):
Found at least 101 unreachable objects in run
```

### Expected behavior:
No error.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
Version: alpha-21196,4840e5fb

#### Windows version:
Windows 10 2004 update

#### Name and version of other software in use when reproducing the issue:
Microsoft® Outlook® Microsoft 365 MSO (16.0.13231.20110) 32-Bit 

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

#### If addons are disabled, is your problem still occuring?
yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
n/a

**Comment by Adriani90** [comment]
cc: @michaelDCurran

**Comment by CyrilleB79** [comment]
That is not the only case when such errors occur.
Are we meant to open an issue for each error of that type:
* if yes, I will try to do it
* if sometimes yes, sometimes no, please indicate in which case we should open an issue
* if no, could we consider to log a warning instead of an error? Indeed, the log would regularly be polluted by error that do not require attention (there are already also the UIA COM errors...)? This would allow to remain focused on real errors that could occur.

**Comment by michaelDCurran** [comment]
For now I would like to see all of these. But we may consider changing 
this from error to warning for the next release.

Coming up to 2020.3 we were able to identify and remove many 
significant  reference cycles which had the potential to freeze NVDA 
when freeing COM objects during a collection.   This particular issue 
still shows there are more situations we need to investigate.

**Comment by CyrilleB79** [comment]
@michaelDCurran , I understand to keep them as an error since they may indicate that there is an issue.
Can you please clarify in which case we should open a new issue when we encounter these errors? Also how can we discriminate garbage collection errors that indicate an underlying problem and those which do not need further investigation?
Thanks.

**Comment by michaelDCurran** [comment]
There is no good rule I'm sorry. I want to know about them all. In a 
perfect world, there should be no garbage collection errors logged.

Perhaps you could open an issue with the first one you see, and then 
keep adding more of them to that one issue. Obviously if they are short 
enough to tell that they are a duplicate of an already reported one 
there then don't bother... but I don't mind the extra noise of having to 
check myself.

**Comment by lukaszgo1** [comment]
@michaelDCurran Just out of curiosity - is the log fragment such as the one from this issue enough for you to understand and fix the reported reference cycle? If so perhaps they should be some documentation on NVDA's wiki on GitHub how to deal with these to allow other developers to fix these. I feell such documentation can be quite helpful for add-on authors who want to avoid garbageHandler warnings in their code.

**Comment by michaelDCurran** [comment]
No, those errors are not really enough to work it out completely. 
Sometimes they are obvious, but the larger they get, the harder it is to 
guess at where it actually started.

The point with these messages is more to alert the developer to the fact 
that there are reference cycles, but not actually what they are. For 
that it takes some pretty complex checking of gc.garbage and 
gc.get_referrers. I will try and find time at some point to write up a 
document on how I do this. But if people are interested, I'd really 
suggest that they just read the documentation for the Python gc module. 
that is how I came up with what I do.

In short though:

1. Instruct gc to save all unreachable objects. In the Python console call:

```

gc.set_debug(gc.DEBUG_SAVEALL)

```

2. Use NVDA for a while and try to reproduce the actions you think might 
cause the reference cycles.

3. If garbage collection errors have not yet been logged, force a 
collect by calling:

```

gc.collect()

```

All unreachable objects will now be stored in gc.garbage. It may be a 
very large list.

To see what other objects refer to an object you can call

```

gc.get_referrers(object)

```

Where object is the object you are interested in.

But again, this is pretty complex stuff, and you need to watch out for 
references that you create simply by checking this stuff. Best to just 
read the Python gc documentation and or other articles on the net about 
this.

**Comment by feerrenrut** [comment]
@Adriani90 Lots of similar situations have been fixed, could you please test this again and report with the issue still occurs?

**Comment by Adriani90** [comment]
This seems fixed in NVDA 2022.4. I am closing.
